packages/next/src/lib/partially-undefined.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * A type that represents a type with all properties of `T` but with `undefined`
+ * values. This is more useful than `Partial<T>` because it allows you to
+ * specify properties that are not optional, and it enforces that all properties
+ * are provided (even if they are `undefined`).
+ */
+export type PartiallyUndefined<T> = {
```

Comment:
It's not a partial anymore though, no? Can't say I have a better word. Not a fan of these utility types anyway. They tend to grow all over the codebase in all sorts of variations. Since they are so underspecified it makes for a harder reading experience due to the added indirection. 

Can we hold of on the abstraction considering it's used in just one place?